HotFix.Specification/Specification.cs

```diff
@@ -47,21 +48,29 @@ public Specification Verify(Action<Session, IConfiguration> verification)
         public Specification Run()
         {
             var clock = new VirtualClock();
-            var transport = new VirtualTransport(clock, Instructions);
+            var transport = new VirtualTransport(clock, Instructions, () => LogoutRequested = true);
             var engine = new Engine { Clocks = c => clock, Transports = c => transport };
 
             var session = (Session) null;
 
             try
             {
-                session = engine.Open(Configuration);
+                using (session = engine.Open(Configuration))
+                {
+                    session.Logon();
 
-                session.Logon();
+                    while (session.Active && !LogoutRequested)
```

Comment:
Is the `LogoutRequested` necessary? It might be simpler for the `VirtualTransport` to be given a reference to the session in order to call `Logout` on it - this would also cause `Active` to be `false` and everything should work as expected...

PS You would need a `if (!Active) return;` at the top of the `Logout` method as it might be called twice...